[rocprofiler-compute] Update Documentation for Strix Halo (#4794)
ROCm/rocm-systems: projects/rocprofiler-compute/docs/conceptual/cdna/cdna-performance-model.rst

.. meta::

>    :description: Understand the AMD CDNA architectures and hardware blocks with ROCm Compute Profiler to analyze and optimize performance on AMD Instinct MI accelerators.

Reviewer: ```suggestion
   :description: Understand the AMD CDNA architectures and hardware blocks with ROCm Compute Profiler to analyze and optimize performance on AMD Instinct MI series GPUs.
```

Reviewer 2: resolved in 206673250825b2c00edf3c245b5c52b2737ff869

   :keywords: ROCm Compute Profiler, CDNA, Instinct, performance model, MI-series

.. _cdna-performance-model:

***********************************
AMD CDNA architecture (CDNA-CDNA4)
***********************************

ROCm Compute Profiler makes available an extensive list of metrics to better understand
achieved application performance on AMD Instinct™ MI-series GPUs

> including CDNA™ architecture based GPUs like the AMD Instinct MI100, CDNA2 architecture based GPUs such as the AMD Instinct MI250, and MI250X, CDNA3 architecture based GPUs such as the AMD Instinct MI300X, MI325X, and CDNA4 architecture based GPUs such as MI350X and MI355X.

Reviewer: ```suggestion
including CDNA™ architecture based GPUs like the AMD Instinct MI100, CDNA2 architecture based GPUs such as the AMD Instinct MI210, MI250, and MI250X, CDNA3 architecture based GPUs such as the AMD Instinct MI300A, MI300X, and MI325X, and CDNA4 architecture based GPUs such as MI350X and MI355X.
```

Reviewer 2: resolved in 206673250825b2c00edf3c245b5c52b2737ff869

The table provides key details and support available for the different CDNA architectures:

✅: Supported
❌: Unsupported

**Architecture details**

.. table::
  :widths: 30 30 30 30 30

  +-----------------+-----------+---------------------------------+-------------------------------------+-------------------------+
  |Architecture     |CDNA       |CDNA 2                           |CDNA 3                               |CDNA 4                   |
  +=================+===========+=================================+=====================================+=========================+
  |Chip packaging   |Single Die |Up to two graphics Compute Dies  |One logical processor with dozen     |Similar to CDNA3,        |
  |                 |           |(GCDs) into single package.      |chiplets, configurable with partition|Multi-Die chiplet, but   |
  |                 |           |                                 |modes.                               |with two I/O Dies (IODs) |
  +-----------------+-----------+---------------------------------+-------------------------------------+-------------------------+

>   |Supported series |MI100      |MI200                            |MI300A                               |MI350X                   |

Reviewer: ```suggestion
  |Supported series |MI100      |MI210                            |MI300A                               |MI350X                   |
```

Reviewer 2: resolved before merge

  |                 |           +---------------------------------+-------------------------------------+-------------------------+

>   |                 |           |MI210                            |MI300X                               |MI355X                   |

Reviewer: ```suggestion
  |                 |           |MI250                            |MI300X                               |MI355X                   |
```

Reviewer 2: resolved before merge

.. meta::
   :description: Understand the AMD CDNA architectures and hardware blocks with ROCm Compute Profiler to analyze and optimize performance on AMD Instinct MI accelerators.
   :keywords: ROCm Compute Profiler, CDNA, Instinct, performance model, MI-series

.. _cdna-performance-model:

***********************************
AMD CDNA architecture (CDNA-CDNA4)
***********************************

ROCm Compute Profiler makes available an extensive list of metrics to better understand
achieved application performance on AMD Instinct™ MI-series GPUs
including CDNA™ architecture based GPUs like the AMD Instinct MI100, CDNA2 architecture based GPUs such as the AMD Instinct MI250X, MI250,

> and MI210, CDNA3 architecture based GPUs such as the AMD Instinct MI300A, MI300X, MI325X, and CDNA4 architecture based GPUs such as MI350X and MI355X.

Reviewer 3: ```suggestion
including CDNA™ architecture based GPUs like the AMD Instinct MI100, CDNA2 architecture based GPUs such as the AMD Instinct MI250, and MI250X, CDNA3 architecture based GPUs such as the AMD Instinct MI300X, MI325X, and CDNA4 architecture based GPUs such as MI350X and MI355X.
```
properly order the entries and align to 2 examples

Reviewer 2: changed this one

The table provides key details and support available for the different CDNA architectures:

✅: Supported
❌: Unsupported

**Architecture details**

.. table::
  :widths: 30 30 30 30 30

  +-----------------+-----------+---------------------------------+-------------------------------------+-------------------------+
  |Architecture     |CDNA       |CDNA 2                           |CDNA 3                               |CDNA 4                   |
  +=================+===========+=================================+=====================================+=========================+
  |Chip packaging   |Single Die |Up to two graphics Compute Dies  |One logical processor with dozen     |Similar to CDNA3,        |
  |                 |           |(GCDs) into single package.      |chiplets, configurable with partition|Multi-Die chiplet, but   |
  |                 |           |                                 |modes.                               |with two I/O Dies (IODs) |
  +-----------------+-----------+---------------------------------+-------------------------------------+-------------------------+

>   |Supported series |MI100      |MI200                            |MI300A                               |MI350X                   |

Reviewer 3: MI200 is not on the list Pratik provided.
MI250 should be added.

Reviewer: Yes. The following are the CDNA2 architecture GPUs that are officially claimed to be supported for ROCm.

<img width="633" height="111" alt="image" src="https://github.com/user-attachments/assets/cb602a71-2777-4911-be33-30dd59d5b510" />

Reviewer 2: resolved in 11f31a720ac0d18e3199c1a0458db2be8a5903cd

.. meta::

>    :description: ROCm Compute Profiler CDNA 2/3/4 (Instinct) performance model — architecture tables and conceptual sections

Reviewer: ```suggestion
   :description: Understand the AMD CDNA architectures and hardware blocks with ROCm Compute Profiler to analyze and optimize performance on AMD Instinct MI accelerators.
```

   :keywords: ROCm Compute Profiler, CDNA, Instinct, performance model, MI-series

.. _cdna-performance-model:

> ************

Reviewer: ```suggestion
***********************************
```

> CDNA 2/3/4

Reviewer: ```suggestion
AMD CDNA architecture (CDNA-CDNA4)
```

> ************

Reviewer: ```suggestion
***********************************
```

ROCm Compute Profiler makes available an extensive list of metrics to better understand

> achieved application performance on AMD Instinct™ MI-series accelerators

Reviewer: ```suggestion
achieved application performance on AMD Instinct™ MI-series GPUs
```

> including Graphics Core Next™ (GCN) GPUs like the AMD Instinct MI50, CDNA™

Reviewer 4: Did we ever support the MI60 at any point? Maybe misremembering.

Generally this is a very long list though, light suggestion to just state in the form of "CDNA2 accelerators form the MI200 series".

Reviewer 3: another suggestion is to use a table here to specify the supports.

Reviewer 2: all MI50 AND MI60 removed

Reviewer 2: I have removed all occurrences of MI50 and MI60

> including Graphics Core Next™ (GCN) GPUs like the AMD Instinct MI50, CDNA™

Reviewer 4: Also why is CDNA the only arch with a trademark on it? We need consistency.

Reviewer: Following our style guide, it is recommended only to use the trademark (™) for the first occurrence on the page.

> including Graphics Core Next™ (GCN) GPUs like the AMD Instinct MI50, CDNA™

Reviewer 3: We dropped support for MI50 and MI60 (see CHANGELOG.md)

Reviewer 2: all MI50 AND MI60 removed

Reviewer 2: removed all MI50 and MI60

> accelerators like the MI100, CDNA2 accelerators such as the AMD Instinct MI250X, MI250,

Reviewer: ```suggestion
architecture based GPUs like the AMD Instinct MI100, CDNA2 architecture based GPUs such as the AMD Instinct MI250X, MI250,
```

> and MI210, CDNA3 accelerators such as the AMD Instinct MI300A, MI300X, MI325X, and CDNA4 accelerators such as MI350X and MI355X.

Reviewer: ```suggestion
and MI210, CDNA3 architecture based GPUs such as the AMD Instinct MI300A, MI300X, MI325X, and CDNA4 architecture based GPUs such as MI350X and MI355X.
```

> For **AMD Radeon™ / RDNA™** client GPUs (e.g. **gfx1151** / RDNA3.5), see

Reviewer: ```suggestion
For AMD Radeon™ / RDNA™ client GPUs (e.g. gfx1151/RDNA3.5), see
```

:doc:`RDNA3 <../rdna/rdna-performance-model>`. The top-level

> :doc:`../performance-model` page summarizes how CDNA and RDNA documentation is organized.

Reviewer: ```suggestion
:doc:`../performance-model` page summarizes how the CDNA and RDNA architecture-specific details are organized.
```

> The table provides key details and support available for the different architectures:

Reviewer: ```suggestion
The table provides key details and support available for the different CDNA architectures:
```

✅: Supported
❌: Unsupported

**Architecture details**

.. table::
  :widths: 30 30 30 30 30

  +-----------------+-----------+---------------------------------+-------------------------------------+-------------------------+
  |Architecture     |CDNA       |CDNA 2                           |CDNA 3                               |CDNA 4                   |
  +=================+===========+=================================+=====================================+=========================+

>   |Chip packaging   |Single Die |Two graphics Compute Dies (GCDs) |One logical processor with dozen     |Similar to CDNA3,        |

Reviewer 4: MI210 only has one GCD

Reviewer 2: solved in commit 6d8d2ca1def7665f59d584d719f5759b46a7e511

Reviewer 4: "The AMD Instinct™ MI200 is built on advanced packaging technologies, enabling two GCDs to
be integrated into a single package in the OAM (OCP Accelerator Module) form factor in the MI250 and MI250X products, as shown in figure
1b. The AMD Instinct MI210 product enables one GCD in a traditional PCIe® form factor card for mainstream HPC and AI workloads in the
data center."
-from the cdna2 whitepaper

There are different GCD counts per MI200 series product, I meant the MI210 has one die, so the original statement of 2 GCDs isn't incorrect, just that mi210 doesn't have 2. We should say "up to 2 GCDs"

Reviewer 2: @cfallows-amd changed to "up to 2 GCDs" in 909594cf7534791f067b57da5fcea8463570a440

>   |Chip packaging   |Single Die |Two graphics Compute Dies (GCDs) |One logical processor with dozen     |Similar to CDNA3,        |

Reviewer 4: For CDNA3 description, depending on which line you read in the whitepaper, it says "a dozen *or more*" chiplets.

Reviewer 2: solved in commit "6d8d2ca1def7665f59d584d719f5759b46a7e511"

  |                 |           |into single package.             |chiplets, configurable with partition|Multi-Die chiplet, but   |

>   |                 |           |                                 |modes.                               |with two I/O Dies (IODs) |

Reviewer 4: Why randomly mentioning IOD count here? There's nothing to compare it to in relation to "CDNA3" when we don't even mention how many IODs CDNA3 can support (up to 4)

Reviewer 2: removed in commit "6d8d2ca1def7665f59d584d719f5759b46a7e511"

  +-----------------+-----------+---------------------------------+-------------------------------------+-------------------------+
  |Supported series |MI100      |MI200                            |MI300A                               |MI350X                   |
  |                 |           +---------------------------------+-------------------------------------+-------------------------+
  |                 |           |MI210                            |MI300X                               |MI355X                   |

>   |                 |           +---------------------------------+-------------------------------------+-------------------------+

Reviewer 4: 308 and 350P?

Reviewer 3: also MI358

Reviewer: Should only include the GPUs that are publicly claimed support for ROCm: https://rocm.docs.amd.com/projects/install-on-linux/en/latest/reference/system-requirements.html#supported-gpus

<img width="831" height="861" alt="Image" src="https://github.com/user-attachments/assets/b8c5d929-48bd-4956-a228-4117a34f9596" />

>   |                 |           |MI250                            |MI325X                               |                         |

Reviewer 3: I think this should be MI250X

Reviewer 2: changed in commit "6d8d2ca1def7665f59d584d719f5759b46a7e511"

  +-----------------+-----------+---------------------------------+-------------------------------------+-------------------------+
  |Spatial partition|❌         |❌                               |Compute partition mode and           |Compute partition mode   |
  |mode             |           |                                 |Memory partition mode                |and Memory partition mode|
  +-----------------+-----------+---------------------------------+-------------------------------------+-------------------------+

**Data type support**

.. list-table::
      :header-rows: 1

      *
        - Architecture
        - FP32
        - FP64
        - FP16
        - INT32 ADD/LOGIC/MAD
        - INT8 DOT
        - INT4 DOT
        - FP32 GEMM
        - FP64 GEMM
        - FP16 GEMM
        - BF16 GEMM
        - INT8 GEMM
        - Packed FP32
        - TF32 GEMM

>         - FP8/BF8

Reviewer 4: GEMM only?

Reviewer 4: To clarify, I think FP8 might be for matrix operations only, it doesn't pass through the VALU

Reviewer 2: @cfallows-amd so you want "FP8/BF8 GEMM"?

Reviewer 2: confirmed that FP8/BF8 only support matrix operation (or GEMM) on strix/halo. I have modified it here on f6a4a98629e723b0b7253b89dbed67fc7cbd5a98

      *
        - CDNA
        - ✅
        - ✅
        - ✅
        - ✅
        - ✅
        - ✅
        - ✅
        - ❌
        - ❌
        - ❌
        - ❌
        - ❌
        - ❌
        - ❌
      *
        - CDNA2
        - ✅
        - ✅
        - ✅
        - ✅
        - ✅
        - ✅
        - ✅
        - ✅
        - ✅
        - ✅
        - ✅
        - ✅
        - ❌
        - ❌
      *
        - CDNA3
        - ✅
        - ✅
        - ✅
        - ✅
        - ✅
        - ✅
        - ✅
        - ✅
        - ✅
        - ✅
        - ✅
        - ✅
        - ✅
        - ✅
      *
        - CDNA4
        - ✅
        - ✅
        - ✅
        - ✅
        - ✅
        - ✅
        - ✅
        - ✅
        - ✅
        - ✅
        - ✅
        - ✅
        - ❌
        - ✅

To best use profiling data, it's important to understand the role of various

> hardware blocks of AMD Instinct accelerators. Refer to the following top level GPU architecture diagram to understand the hardware blocks of each architectures.

Reviewer: ```suggestion
hardware blocks of AMD Instinct GPUs. Refer to the following top-level GPU architecture diagram to understand the hardware blocks of each architecture.
```

.. tab-set::

  .. tab-item:: CDNA

    .. image:: ../../data/conceptual/CDNA.png
      :alt: CDNA top level architecture diagram with zoomed view of Compute unit

  .. tab-item:: CDNA2

    .. image:: ../../data/conceptual/CDNA2.png
      :alt: CDNA2 top level architecture diagram with zoomed view of Compute unit

  .. tab-item:: CDNA3

    .. image:: ../../data/conceptual/CDNA3.png
      :alt: CDNA3 top level architecture diagram with zoomed view of Accelerator Complex Dies (XCDs)

  .. tab-item:: CDNA4

    .. image:: ../../data/conceptual/CDNA4.png
      :alt: CDNA4 top level architecture diagram

> This section describes each hardware block on the accelerator as interacted with by a software developer to

Reviewer: ```suggestion
This section describes each hardware block on the GPUs as interacted with by a software developer to
```

> give a deeper understanding of the metrics reported by profiling data. Refer to

Reviewer 3: This run-on sentence is pretty hard to digest.

:doc:`/tutorial/profiling-by-example` for more practical examples and details on how
to use ROCm Compute Profiler to optimize your code.

.. _mixxx-note:

.. note::

>    In this documentation, **MI2XX** refers to any of the CDNA2 architecture-based MI200 series accelerators such as AMD

Reviewer: ```suggestion
   In this documentation, **MI2XX** refers to any of the CDNA2 architecture-based MI200 series GPUs, such as AMD
```

>    Instinct MI250X, MI250, and MI210 accelerators interchangeably in cases

Reviewer: ```suggestion
   Instinct MI250X, MI250, and MI210 GPUs interchangeably in cases
```

   where the exact product at hand is not relevant. For product details, see `AMD Instinct GPUs <https://www.amd.com/en/products/accelerators/instinct.html>`_.

>    For a comparison of AMD Instinct accelerator specifications, refer to

Reviewer: ```suggestion
   For a comparison of AMD Instinct GPU specifications, refer to
```
